### ISSUE TITLE
Vertical lines through all content

### ISSUE THREAD

**Issue by jyapayne** [browsh-org/browsh#77]
I'm on Ubuntu 18.04 x64 and just installed your Ubuntu build from the website. Upon executing `browsh` in a gnome-terminal, I get this picture:

![screenshot from 2018-07-10 10-42-37](https://user-images.githubusercontent.com/2645657/42484442-031b223c-842e-11e8-9754-80a67d1e6032.png)

Not sure what's wrong, but I can't really view anything. Tried searching, and it remains like this.

Edit: exact version of browsh is the latest: 1.2.2

**Comment by tombh** [comment]
Oh wow, I've not seen this particular visual artefact before. Can you try a different font in the same terminal and also try another terminal. This'll just narrow down the problem, thanks.

**Comment by jyapayne** [comment]
Hmm, different fonts seem to have no effect. I tried Ubuntu Mono Regular and the default font for gnome-terminal. My firefox version is 61.0 if that helps. I tried terminator as well, but I still get the same render. What terminal do you use?

**Comment by crubier** [comment]
Horizontal lines on mac os !

<img width="1111" alt="screen shot 2018-07-09 at 22 44 37" src="https://user-images.githubusercontent.com/2954572/42491491-cb027a80-83c9-11e8-9cb8-fa9f94eadf9e.png">

**Comment by tobimensch** [comment]
I had similar issues and it seems those can be fixed by either using a suitable terminal (i.e. Konsole) and/or the latest mosh from git.

**Comment by jyapayne** [comment]
@tobimensch right, but I don't like either of those :P Plus, I think it would be nice if this supported the default terminal in Ubuntu so that people can just try it out without having to install too many things.

**Comment by tobimensch** [comment]
@jyapayne 
For that to work browsh probably needs a non-true color mode. Ie.e 256 and/or 16 color mode.

**Comment by tombh** [comment]
For those that don't want to setup true colour, for now you can just toggle monochrome mode: `ALT+m`.

Although I would like to see Browsh support less colours than true colour, it's not a high priority at the moment.

**Comment by jyapayne** [comment]
@tobimensch but gnome-terminal on Ubuntu 18.04 does support true color. 
@tombh I just tried Konsole, and it has a similar issue for me.

![screenshot from 2018-07-10 20-26-56](https://user-images.githubusercontent.com/2645657/42507316-ad63a1fa-847f-11e8-8402-6eb42be21702.png)

**Comment by tobimensch** [comment]
@jyapayne:
Compile the latest mosh from github and try with that instead of ssh.

**Comment by jyapayne** [comment]
@tobimensch but I'm not using ssh. I'm just on my local machine

**Comment by tobimensch** [comment]
@jyapayne 
I'm only ever using browsh from a remote server. I'm not sure what's causing those distortions you're seeing.

**Comment by jyapayne** [comment]
I'd really like to use this in order to have a browser in my terminal because I work mostly in vim with tmux and it gets annoying to have to switch to chrome/firefox to google things. Maybe boot up an Ubuntu VM and see if the same behavior is observed?

**Comment by tobimensch** [comment]
@jyapayne 
Good idea. I'm hosting browsh on Debian Jessie.

**Comment by Determinant** [comment]
Same problem on Archlinux.

**Comment by danielcb** [comment]
also on gentoo/urxvt

**Comment by ghostseven** [comment]
Same here Ubuntu 16.04.4 latest version of browsh

**Comment by tombh** [comment]
@Determinant @danielcb and @ghostseven: are you sure you have the same visual artefacts? Because @jyapayne has *vertical* lines, which is unusual. Whereas it's more typical to see *horizontal* lines like @crubier's screenshot above: https://github.com/browsh-org/browsh/issues/77#issuecomment-403707407 These are certainly caused by lack of true colour support.

@jyapayne I'm still at a loss as to what's causing your problem. The only thing I can think is to keep cycling through fonts and terminals. I'm using Alacritty on Arch with Ubuntu Mono and I know Source Code Pro works as well. If after trying lots of fonts and terminals you still have the same issue, then that at least is an important data point to have.

**Comment by ghostseven** [comment]
My apologies I am having the horizontal lines issue not vertical, I should have clarified this.

**Comment by Determinant** [comment]
Also sorry that I'm seeing the horizontal lines instead of vertical ones.

**Comment by jyapayne** [comment]
@tombh I finally found what is wrong! After deleting lines in my bashrc to see if anything changes, the offending line is this one:

```bash
export LANG="ja_JP.UTF-8"
```

If I remove that, everything is fine. I had it in my bashrc for some Wine configuration of Japanese applications.

**Comment by tombh** [comment]
@Determinant so using the same terminal (256 colours) the SSH demo looks ok, but local Browsh has horizontal lines? Browsh doesn't actually formally support anything other than true colour and monochrome mode (`ALT+m`). So I'm interested to hear that you're having some success with a 256 colour terminal.

@jyapayne Nice detective work! Well that's good to have in the archives here, in case someone comes across it in the future. I'll close this now then.

**Comment by happyme531** [comment]
export LANG="en_US.UTF-8" before launching browsh  
fixed the issue. my LANG is zh_CN.UTF-8 by default.  
tested on vscode builtin terminal and windows terminal. works fine.

**Comment by soshimee** [comment]
I see exactly same issues, except that it is not caused by LANG.

**Comment by tombh** [comment]
@soshimee and you've definitely confirmed that your terminal supports true colour? 

https://gist.github.com/XVilka/8346728

**Comment by soshimee** [comment]
Yes, I tried the code, and I see smooth rainbow. I'm using Konsole to be specific.

**Comment by tombh** [comment]
oh that's really interesting. and have you tried different fonts. if you could add a screenshot, it could help others too

**Comment by soshimee** [comment]
![Screenshot_20210902_091114](https://user-images.githubusercontent.com/45235204/131761315-676fd513-edbe-401d-a8d6-296e870f0cde.png)

**Comment by soshimee** [comment]
It does disappear after scrolling, but letters become even more scrambled.

**Comment by tombh** [comment]
thanks. and have you tried different fonts?

**Comment by soshimee** [comment]
Different fonts don't solve the issue.

**Comment by tombh** [comment]
what about a font that specifically has the "UTF-8 Half Block" in it? https://www.fileformat.info/info/unicode/char/2584/index.htm

i think if a font doesn't have it falls back to another, that fall back font might be your problem? i usually just have google's noto fonts on my system for example

**Comment by soshimee** [comment]
There is that unicode character, it's that Browsh is not refreshing every character but only every 2.

**Comment by tombh** [comment]
it's unlikely to be browsh doing that, some terminals don't render.... ohhhhhhhhhhh!!!!!!!!!!!!! suddenly i understand, hangul characters are double width 🥳 ok so you need a terminal or some setting in konsole, that doesn't break line width when rendering a mix of single and double width chars

**Comment by soshimee** [comment]
It does that even in english sites, and you can see the browsh booting screen behind the google.

**Comment by soshimee** [comment]
With Alacritty it seems to display checkerboard pattern.

**Comment by tombh** [comment]
can you send a screenshot?

**Comment by soshimee** [comment]
![Screenshot_20210902_123048](https://user-images.githubusercontent.com/45235204/131777209-18386447-06bd-424f-ba19-f25e3ab7c267.png)

**Comment by tombh** [comment]
i really feel like that is the font. the utf8 half block is misaligned

**Comment by soshimee** [comment]
I don't know how to change font on Alacritty.

**Comment by tombh** [comment]
i think it's your system's fallback font, so it's not set in alacritty or konsole. that's why i'd recommend installing google noto fonts, just because it's such a comprehensive font

**Comment by soshimee** [comment]
It is already installed since I deal with unicodes a lot.

**Comment by tombh** [comment]
hmm, then i've run out of ideas. the only thing i can suggest is to explore how fallback fonts are setup on your system. sorry 😕

**Comment by soshimee** [comment]
Kitty terminal seems to have the same issue, but the same with Konsole.

**Comment by soshimee** [comment]
Any updates?

**Comment by tombh** [comment]
sorry no, i can't look into this :( all i can say is that i think it's your fallback font

**Comment by soshimee** [comment]
It doesn't look like font that is...

**Comment by soshimee** [comment]
On macOS, I can confirm this issue as well running on iTerm.

**Comment by soshimee** [comment]
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/45235204/133626968-b640667e-f015-480d-a697-685a824f6d3d.png">

**Comment by Alumniminium** [comment]
I have this issue too, but only if I use mosh, it renders correctly using ssh and locally.

**Comment by tombh** [comment]
Are you using the unofficial main branch build of Mosh? If no then you won't have true colour support, and so will quite likely suffer from these lines

**Comment by soshimee** [comment]
No, running locally.

**Comment by persakiss** [comment]
On M1 Macbook Air, same issue, Monterey, using iTerm 2.

**Comment by tombh** [comment]
Could you screenshot the true colour test?

**Comment by persakiss** [comment]
> Could you screenshot the true colour test?
<img width="569" alt="Screen Shot 2022-06-15 at 4 00 22 PM" src="https://user-images.githubusercontent.com/12379641/173822344-039f01d5-301b-4440-98ec-6f58292b8f9b.png">

Output looks to pass the true color test(?), but $colorterm and echo $term do not show truecolor.

**Comment by soshimee** [comment]
My linux machine was wiped due to its motherboard and RAM breaking down, and technician thinking that it was with my SSD. But I can still send one from my macOS machine, since it's still working. I don't know why but browsh won't work on it anymore though. Still, I'll send you the test.
<img width="682" alt="image" src="https://user-images.githubusercontent.com/45235204/173845724-ffe42f4b-eb75-41f1-9b2a-57208fb93a47.png">
If this isn't the test you meant, please tell me!

**Comment by tombh** [comment]
Oh! @persakiss deleted their comments!??

@soshimee I wasn't replying to you. I think we already established that your issue is due to the Korean characters

**Comment by soshimee** [comment]
It isn't. It displays that even on english-only websites and the characters are actually not changing. They are left behind.

**Comment by tombh** [comment]
So do you have any ideas of what may be the underlying cause?

**Comment by soshimee** [comment]
The only thing I can think of is browsh itself only printing one every two characters.

**Comment by tombh** [comment]
Of course, but why would it do that? It's not a general bug, it's specific to your setup. We at least need a theory about what's different about your setup. The theory doesn't have to be right, it just has to be a theory

**Comment by persakiss** [comment]
> Oh! @persakiss deleted their comments!??
> 
> @soshimee I wasn't replying to you. I think we already established that your issue is due to the Korean characters

Sorry about that bud, I meant to edit my comment but it deleted the first one, which gave no context to my second so I deleted that too.

To bring the context back, I'm running a M1 Macbook Air, which should have truecolor.. I've tried iTerm 2, Hyper, Termius, Kitty, all give the same issue (horizontal lines on Browsh running on a remote server).

<img width="569" alt="Screen Shot 2022-06-15 at 4 00 22 PM" src="https://user-images.githubusercontent.com/12379641/174667998-06cf0dd5-1f2b-44db-b31c-13f6b2d3f3cb.png">

<img width="589" alt="Screen Shot 2022-06-14 at 11 22 42 PM" src="https://user-images.githubusercontent.com/12379641/174668011-19c0f2be-989a-4395-8275-85985e29675b.png">

**Comment by soshimee** [comment]
I don't know, maybe browsh doesn't like Korean locale? Not the characters themselves, but the locale.

**Comment by tombh** [comment]
@persakiss are you using Browsh through MoSH there? If so then you have to be using the master branch version, as the current offcial release of mosh doesn't suppport true colour

**Comment by wtdcode** [comment]
I found having this issue when using `screen` windows...

**Comment by yungcrypt** [comment]
I had this issue on Mac OS but when i run it in tmux it looks fine.

**Comment by zeroooi** [comment]
<img width="1165" alt="Image" src="https://github.com/user-attachments/assets/7e75b5d8-2103-4d41-bec0-18585c10745c" />
如果你使用item2，打开这个选项可以解决问题

**Comment by jerivl** [comment]
Fixed on Ghostty with https://github.com/ghostty-org/ghostty/issues/8709
Update to >1.2.1 (nixpkgs25.05 is on 1.1.3)
